### ISSUE TITLE
use JSHint and CSSLint

### ISSUE THREAD

**Issue by TitanNano** [loqui/im#590]
I think it would be better if everyone would use JSHint and CSSLint. I use them by default and they throw many many errors. If we also use Travis CI we can add a JSHint and a CSSLint test. With that we probably could get better code and also faster CSS rules...

**Comment by TitanNano** [comment]
If everyone agrees, I would lint all the js files and then set up a pre-commit hook based on [this](http://stackoverflow.com/questions/15703065/github-setup-pre-commit-hook-jshint/26557272#26557272). So we would have some cleaner JavaScript.

**Comment by Gioyik** [comment]
Me parece bien. Sin embargo hay que tomar en cuenta para el caso de JSHint agregar las excepciones que sean necesarias en algunos warnings y errores. No todo lo que detecte JSHint significa que este mal. :)

**Comment by TitanNano** [comment]
@Gioyik would it be possible to provide an English translation? :)

**Comment by Gioyik** [comment]
Sorry, I was thinking on Spanish while I write the answer.

It's ok for me (I think is a must have and it will avoid future issues). However remember to add exceptions for JSHint warnings and errors in Javascript code. Not all the things that JSHint points as bad or bad-formed are true cases of issues. :)

**Comment by TitanNano** [comment]
yes of course. We will have a .jshintrc file where we can disable any rule we don't like :)

**Comment by Gioyik** [comment]
great, feel free to ping if you need help.
